eng/pipelines/templates/jobs/vmr-synchronization.yml

```diff
@@ -25,12 +25,16 @@ jobs:
   - template: /eng/common/templates/variables/pool-providers.yml
   - name: vmrPath
     value: $(Agent.BuildDirectory)/vmr
-  - ${{ if eq(variables['System.TeamProject'], 'internal') }}:
+  - ${{ if and( eq(variables['System.TeamProject'], 'internal'), or(startswith(variables['Build.SourceBranch'], 'refs/heads/release/'), eq(variables['Build.SourceBranch'], 'refs/heads/main'))) }}:
     - group: DotNetBot-GitHub
-  - ${{ else }}:
-    - name: BotAccount-dotnet-bot-repo-PAT
-      value: N/A
-
+    - group: DotNetBot-GitHub-No-Scopes
+  - ${{ if and( eq(variables['System.TeamProject'], 'internal'), startswith(variables['Build.SourceBranch'], 'refs/heads/internal/release/')) }}:
+    - group: DotNetBot-AzDO-PAT
+  - name: vmrPublicUrl
+    value: https://github.com/dotnet/dotnet
+  - name: vmrPrivateUrl
+    value: https://dnceng@dev.azure.com/dnceng/internal/_git/dotnet-dotnet
```

Comment:
Apologies, one last idea - can we define these variables only when they should be used to also make sure we don't use the wrong one below?

Reply:
sure, I can put them with the corresponding variable group